libs/angular-routing/src/lib/link-active.directive.ts

```diff
@@ -81,16 +82,20 @@ export class LinkActive implements AfterContentInit, OnDestroy, OnChanges {
           .map((link) =>
             link.hrefUpdated.pipe(
               startWith(link.linkHref),
-              mapTo(link.linkHref)
+              mapTo(link.linkHref),
+              filterNullable()
```

Comment:
This can be replaced with
```
filter(href => href != null)
```
(notice the comparison by value instead of reference). It avoids using an external operator.

Otherwise, if we need to have an external operator I would rather wrap all three operators - startWith, mapTo and filter nullable to extract the common functionality

```
export const processLink = <LinkTo>(link: LinkTo) => (source: Observable<LinkTo>): OperatorFunction<LinkTo, string> =>
  source.pipe(
    startWith(link.linkHref),
    mapTo(link.linkHref),
    filter(href => href != null)
  );
```

Reply:
Hi @meeroslav,

Thanks for reviewing.

The main reason why I did extract it is to provide type guards which really start to pay off when strictNullChecks are enabled https://github.com/microsoft/TypeScript/issues/16069. I would prefer to keep current behavior also due to the fact that even though we can do `href != null`, it feels more descriptive to provide strict checks for `null` and `undefined`.
I would also like to take your idea of wrapping it in a custom `processLink` operator.

Please let me know if this is acceptable.